### PR TITLE
[FEATURE] Refactor wizard type=suggest to suggestOptions

### LINKED ISSUE #1990: Migrate suggest wizard type group
https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/8.6/Deprecation-79440-TcaChanges.html

Basically write a rector for this method https://github.com/TYPO3/TYPO3.CMS/blob/9e1c91aed86490f75730ae186504bd900057107e/typo3/sysext/core/Classes/Migrations/TcaMigration.php#L2138

### PR DESCRIPTION
**Pull request by MK-42**
closes #1990

Some refactoring done in TcaHelperTrait. Feel free to comment about the style of building up the rector here - This would be a first draft of a (hopefully) more readable tca Rector style. What do you think?

I would vote for removing some of the convenience methods in the trait, for better visibility of stuff like 
```
            if (! $this->hasKeyValuePair($config, 'type', 'group')) {
```

### CODE REVIEW

**Comment by MK-42** [comment]
I'm unsure if i read the core migration regarding hideSuggest correctly. Can anyone contribute a real-life example how such a hideSuggest config in ColumnsOverrides would look like, before/after?

@sabbelasichon what do you think about this style of TCA-Rector writing? Should I try to streamline some more of them? Or would you prefer some other tweaks / readability enhancements?

File: src/Rector/v8/v6/MoveTypeGroupSuggestWizzardToSuggestOptionsRector.php [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+use Rector\Core\Rector\AbstractRector;
+use Ssch\TYPO3Rector\Helper\TcaHelperTrait;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/8.6/Deprecation-79440-TcaChanges.html#suggest-wizard
+ * @see \Ssch\TYPO3Rector\Tests\Rector\v8\v6\MoveTypeGroupSuggestWizzardToSuggestOptions\MoveTypeGroupSuggestWizzardToSuggestOptionsTest
+ */
+final class MoveTypeGroupSuggestWizzardToSuggestOptionsRector extends AbstractRector
```
Comment (sabbelasichon) [inline src/Rector/v8/v6/MoveTypeGroupSuggestWizzardToSuggestOptionsRector.php:21]: Should it be Wizard instead of Wizzard

**Comment by sabbelasichon** [comment]
Thank you very much. That looks a way more readable than before.

**Comment by sabbelasichon** [comment]
> I'm unsure if i read the core migration regarding hideSuggest correctly. Can anyone contribute a real-life example how such a hideSuggest config in ColumnsOverrides would look like, before/after?

Maybe the test are a good starting point: https://github.com/TYPO3/TYPO3.CMS/blob/9e1c91aed86490f75730ae186504bd900057107e/typo3/sysext/core/Tests/Unit/Migrations/TcaMigrationTest.php#L5153

**Comment by MK-42** [comment]
> > I'm unsure if i read the core migration regarding hideSuggest correctly. Can anyone contribute a real-life example how such a hideSuggest config in ColumnsOverrides would look like, before/after?
> 
> Maybe the test are a good starting point: https://github.com/TYPO3/TYPO3.CMS/blob/9e1c91aed86490f75730ae186504bd900057107e/typo3/sysext/core/Tests/Unit/Migrations/TcaMigrationTest.php#L5153

honestly I didn't think about the possibility that the core would have tests for the migrations. I've added them all to the suite and adapted the rector to discard empty configurations

File: src/Rector/v8/v6/MoveTypeGroupSuggestWizzardToSuggestOptionsRector.php (hunk 1 above)
Comment (MK-42) [reply src/Rector/v8/v6/MoveTypeGroupSuggestWizzardToSuggestOptionsRector.php]: of course it should :)

**Comment by MK-42** [comment]
I think I'm done for now here - this should be it for this migration. feel free to review!

Sorry for mangling some refactoring into this feature. I hope its okay to leave these changes in here - otherwise i can pick the unrelated commits onto another branch.

**Comment by sabbelasichon** [comment]
Thanks a bunch. That´s just amazing.

**Comment by sabbelasichon** [comment]
@MK-42 I know why i asked you. Very well done. You improved the shitty code to a better. Thanks a lot.

**Pull request merged**